undefined reference to usrsctp_get_timeout

build fails with wrong version of sctplab

expected: build should fail early in configure phase

`usrsctp_get_timeout` is added in https://github.com/sctplab/usrsctp/pull/591

https://github.com/frida/frida-core/blob/ab2581a0b7faea22d87a33c1cc6f2aab8bd88ece/lib/base/p2p-glue.c#L347

error:

```
[57/144] Linking target lib/agent/libfrida-agent.so
FAILED: lib/agent/libfrida-agent.so 
g++  -o lib/agent/libfrida-agent.so ...
ld: lib/base/libfrida-base-1.0.a(p2p-glue.c.o): in function `_frida_sctp_timer_source_get_timeout':
p2p-glue.c:(.text+0x8b3): undefined reference to `usrsctp_get_timeout'
```

--- a/meson.build
+++ b/meson.build
@@ -231,9 +231,20 @@ if get_option('b_sanitize') == 'address'
 endif
 
 glib_dep = dependency('glib-2.0', version: '>=2.56')
+gio_dep = dependency('gio-2.0')
+
+assert(
+  # vala: Thread.garbage_collect
+  cc.has_function('g_thread_garbage_collect', dependencies: [glib_dep]) and
+  # vala: GLibFork.prepare_to_fork
+  cc.has_function('glib_prepare_to_fork', dependencies: [glib_dep]) and
+  # vala: GIOFork.prepare_to_fork
+  cc.has_function('gio_prepare_to_fork', dependencies: [gio_dep]),
+  'please use vala from https://github.com/frida/vala and glib from https://github.com/frida/glib'
+)
+
 gobject_dep = dependency('gobject-2.0')
 gmodule_dep = dependency('gmodule-2.0')
-gio_dep = dependency('gio-2.0')
 gee_dep = dependency('gee-0.8')
 json_glib_dep = dependency('json-glib-1.0')
 libsoup_dep = dependency('libsoup-3.0')

--- a/meson.build
+++ b/meson.build
@@ -292,6 +292,11 @@ if nice_dep.found()
                    prefix: '#include <usrsctp.h>')
     cdata.set('HAVE_SCONN_LEN', 1)
   endif
+
+  assert(
+    cc.has_function('usrsctp_get_timeout', dependencies: [usrsctp_dep]),
+    'please use usrsctp with https://github.com/sctplab/usrsctp/pull/591'
+  )
 else
   openssl_dep = []
   usrsctp_dep = []

--- a/meson.build
+++ b/meson.build
@@ -309,7 +309,7 @@ if nice_dep.found()
 
   assert(
     cc.has_function('usrsctp_get_timeout', dependencies: [usrsctp_dep]),
-    'please use usrsctp with https://github.com/sctplab/usrsctp/pull/591'
+    'Please use usrsctp with https://github.com/sctplab/usrsctp/pull/591'
   )
 else
   openssl_dep = []

--- a/meson.build
+++ b/meson.build
@@ -240,7 +240,7 @@ assert(
   cc.has_function('glib_prepare_to_fork', dependencies: [glib_dep]) and
   # vala: GIOFork.prepare_to_fork
   cc.has_function('gio_prepare_to_fork', dependencies: [gio_dep]),
-  'please use vala from https://github.com/frida/vala and glib from https://github.com/frida/glib'
+  'Please use Vala from https://github.com/frida/vala and GLib from https://github.com/frida/glib'
 )
 
 gobject_dep = dependency('gobject-2.0')

--- a/meson.build
+++ b/meson.build
@@ -282,7 +282,10 @@ if host_os == 'android'
   backend_deps_private += [libselinux_dep, libsepol_dep]
 endif
 
-tls_provider_dep = dependency('gioopenssl', required: get_option('connectivity'))
+tls_provider_dep = dependency('gioopenssl',
+  required: get_option('connectivity'),
+  not_found_message: 'please use glib-networking from https://github.com/frida/glib-networking'
+)
 if tls_provider_dep.found()
   cdata.set('HAVE_GIOOPENSSL', 1)
   vala_flags += ['--define=HAVE_GIOOPENSSL']

--- a/meson.build
+++ b/meson.build
@@ -284,7 +284,7 @@ endif
 
 tls_provider_dep = dependency('gioopenssl',
   required: get_option('connectivity'),
-  not_found_message: 'please use glib-networking from https://github.com/frida/glib-networking'
+  not_found_message: 'Please use GLib Networking from https://github.com/frida/glib-networking'
 )
 if tls_provider_dep.found()
   cdata.set('HAVE_GIOOPENSSL', 1)

